lib/soapy_cake/admin_addedit.rb

```diff
@@ -173,6 +172,15 @@ def edit_campaign(opts)
 
     private
 
+    def addedit_offer_tier(add_edit_option, opts)
+      require_params(opts, %i(offer_id tier_id price_format_id offer_contract_id status_id))
+
+      opts = opts.merge(redirect_offer_contract_id: -1, add_edit_option: add_edit_option)
```

Comment:
Hmm, then we should either not allow passing in the same object twice or not modify the objects anywhere in soapy_cake. This seems very inconsistent.